pkg/controller/kubedirectorcluster/members.go

```diff
@@ -509,6 +510,95 @@ func handleCreatePendingMembers(
 	wgRunning.Wait()
 }
 
+// handleConfigPackage makes the backup of /opt/guestconfig folder when the pod is upgrading
+// and restores /opt/guestconfig folder from the backup when the pod is rolling back
+func handleConfigPackage(
+	reqLogger logr.Logger,
+	cr *kdv1.KubeDirectorCluster,
+	member *kdv1.MemberStatus,
+) {
+
+	// Do nothing if the pod is not upgrading/rolling back
+	if member.PodUpgradeStatus != kdv1.PodUpgrading && member.PodUpgradeStatus != kdv1.PodRollingBack {
+		return
+	}
+
+	logErrWrapper := func(err error, message string) {
+		shared.LogError(
+			reqLogger,
+			err,
+			cr,
+			shared.EventReasonMember,
+			message)
+	}
+
+	pod, podErr := observer.GetPod(cr.Namespace, member.Pod)
+	if podErr != nil {
+		logErrWrapper(podErr, fmt.Sprintf("failed to get pod %s", member.Pod))
+	}
+
+	execCommandWrapper := func(cmd string) error {
+
+		var stdOut bytes.Buffer
+		ioStreams := &executor.Streams{Out: &stdOut}
+
+		return executor.ExecCommand(
```

Comment:
This is a fine way to run the command I think, but it also seems pretty similar to what executor.RunScript does... I'm curious if you tried that and there was a problem with it.

BTW if we do want/need to do it this way instead of using RunScript, then we should probably export executor.execShell (i.e. turn it into executor.ExecShell) and use that here instead of hardcoding "/bin/bash". I realize we do have /bin/bash hardcoded in a couple of other places too, but we'll need to deal with all of those if/when we tackle issue #402 .

Reply:
Yes, `RunScript` works ok, fixed